bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/DeviceInspector.java

```diff
@@ -140,16 +258,28 @@ private Map<String, String> readDeviceDescription(IndividualAddress address) {
         Map<String, String> ret = new HashMap<>();
         byte[] data = getClient().readDeviceDescription(address, 0, false, OPERATION_TIMEOUT);
         if (data != null) {
-            final DD0 dd = DeviceDescriptor.DD0.from(data);
-
-            ret.put(FIRMWARE_TYPE, Firmware.getName(dd.firmwareType()));
-            ret.put(FIRMWARE_VERSION, Firmware.getName(dd.firmwareVersion()));
-            ret.put(FIRMWARE_SUBVERSION, Firmware.getName(dd.firmwareSubcode()));
-            logger.debug("The device with address {} is of type {}, version {}, subversion {}", address,
-                    Firmware.getName(dd.firmwareType()), Firmware.getName(dd.firmwareVersion()),
-                    Firmware.getName(dd.firmwareSubcode()));
+            try {
+                final DD0 dd = DeviceDescriptor.DD0.from(data);
+
+                ret.put(DEVICE_MASK_VERSION, String.format("%04X", dd.maskVersion()));
+                ret.put(DEVICE_PROFILE, dd.deviceProfile());
+                ret.put(DEVICE_MEDIUM_TYPE, getMediumType(dd.mediumType()));
+                logger.debug("The device with address {} has mask {} ({}, medium {})", address,
+                        ret.get(DEVICE_MASK_VERSION), ret.get(DEVICE_PROFILE), ret.get(DEVICE_MEDIUM_TYPE));
+            } catch (KNXIllegalArgumentException e) {
+                logger.info("Can not parse Device Descriptor 0 of device with address {}: {}", address, e.getMessage());
+            }
         } else {
-            logger.debug("The KNX device with address {} does not expose a Device Descriptor", address);
+            logger.debug("The device with address {} does not expose a Device Descriptor type 0", address);
+        }
+        data = getClient().readDeviceDescription(address, 2, false, OPERATION_TIMEOUT);
+        if (data != null) {
+            try {
+                final DD2 dd = DeviceDescriptor.DD2.from(data);
+                logger.debug("The device with address {} is has DD2 {}", address, dd.toString());
```

Comment:
All this block is only for logging a data, no property is set ?

Reply:
Actually I have not seen any device exposing a DD2 descriptor yet. The KNX spec says that each device needs to implement either DD0 or DD2. Maybe this log entry helps to diagnose once we get a complaint about devices which do not show data.

Reply:
So I would suggest to add all this block inside a if block based on condition logger.isDebugEnabled().
Like that, it will not be run for nothing when DEBUG logs is not enabled.